backend/apps/webui/main.py

```diff
@@ -243,43 +246,33 @@ def get_pipe_id(form_data: dict) -> str:
     return pipe_id
 
 
-def get_function_params(function_module, form_data, user, extra_params={}):
+def get_function_params(function_module, form_data, user, extra_params=None):
+    if extra_params is None:
+        extra_params = {}
+
     pipe_id = get_pipe_id(form_data)
     # Get the signature of the function
     sig = inspect.signature(function_module.pipe)
-    params = {"body": form_data}
-
-    for key, value in extra_params.items():
-        if key in sig.parameters:
-            params[key] = value
+    addition_params = {k: v for k, v in extra_params.items() if k in sig.parameters}
+    params = {"body": form_data} | addition_params
 
-    if "__user__" in sig.parameters:
-        __user__ = {
-            "id": user.id,
-            "email": user.email,
-            "name": user.name,
-            "role": user.role,
-        }
+    if hasattr(function_module, "UserValves"):
+        user_valves = Functions.get_user_valves_by_id_and_user_id(pipe_id, user.id)
+        params["__user__"]["valves"] = function_module.UserValves(
```

Comment:
By removing try: except here this could throw if `user_valves` is incompatible with `function_module.UserValves`